src/gt4sd/algorithms/core.py

```diff
@@ -442,6 +443,34 @@ def list_versions(cls) -> Set[str]:
         versions = versions.union(get_algorithm_subdirectories_in_cache(prefix))
         return versions
 
+    # QUESTION: should this be in core.py? Uploading to S3 is not a core functionality.
```

Comment:
in principle I agree with the comment, but listing remote versions I think it's useful. Maybe we should rename it thinking about the future and the possibility of using other types of remote storage systems. My main question here is, how this differ from `list_versions` besides the function signature?
If they do not introduce any change in the logic I would call one inside the other to reduce code duplication.

Reply:
In `list_versions` we search on s3 (`get_algorithm_subdirectories_with_s3`) and locally (`get_algorithm_subdirectories_in_cache`) for versions. 
In `list_versions_s3` we search only s3 because we could have already saved our version in the cache (for example using cli-saving before cli-upload) but still want it on s3.

We can call one inside the other but then we need two outputs: one set with the versions on s3 and another set with all the versions (s3 and cached).

Reply:
ah good catch sorry I missed it, then I would simply rename `list_versions_s3` to `list_remote_versions`.